core/src/test/java/org/elasticsearch/index/engine/InternalEngineTests.java

```diff
@@ -2615,10 +2615,13 @@ public void testHandleDocumentFailure() throws Exception {
                 }
                 Engine.IndexResult indexResult = engine.index(indexForDoc(doc1));
                 assertNotNull(indexResult.getFailure());
-
+                // document failures should be recorded in translog
+                assertNotNull(indexResult.getTranslogLocation());
```

Comment:
commenting here because this is the only change in this file. Shouldn't we have some test around the noop used on replicas?